tests/common/test_logger.py

```diff
@@ -161,34 +174,336 @@ def test_periodic_forwards_message_and_args(self, mock_info, mock_error, mock_wa
+        self.assertIn(child_prefix, mock_telem_write.call_args[0][1])
+        self.assertIn(child_prefix, mock_stdout_write.call_args[0][1])
+
+        logger.error("Test Log")
+        self.assertEqual(2, mock_file_write.call_count)
+        self.assertEqual(2, mock_console_write.call_count)
+        self.assertEqual(2, mock_telem_write.call_count)
+        self.assertEqual(2, mock_stdout_write.call_count)
+
+        self.assertIn(parent_prefix, mock_file_write.call_args[0][1])
+        self.assertIn(parent_prefix, mock_console_write.call_args[0][1])
+        self.assertIn(parent_prefix, mock_telem_write.call_args[0][1])
+        self.assertIn(parent_prefix, mock_stdout_write.call_args[0][1])
+
+    @patch("azurelinuxagent.common.event.send_logs_to_telemetry", return_value=True)
+    @patch('azurelinuxagent.common.logger.Logger.error')
+    @patch('azurelinuxagent.common.logger.Logger.warn')
+    def test_telemetry_logger_verify_not_logging_errors_warnings(self, mock_warn, mock_error, *args):
+        appender = logger.TelemetryAppender(logger.LogLevel.WARNING, add_log_event)
+
+        with patch('azurelinuxagent.common.event.EventLogger.save_event') as mock_save:
+            appender.write(logger.LogLevel.WARNING, 'Cgroup controller "memory" is not mounted. '
+                                                    'Microsoft.OSTCExtensions.DummyExtension-1.2.3.4')
+            self.assertEqual(1, mock_save.call_count)
+            self.assertEqual(0, mock_warn.call_count)
+            self.assertEqual(0, mock_error.call_count)
+
+        # Writing 2000 events should generate only one more log event due to too many files. #1035 was caused due to
+        # too many files being written in an error condition (earlier code would write 1000 files in this case).
+        for _ in range(2000):
+            appender.write(logger.LogLevel.WARNING, 'Cgroup controller "memory" is not mounted. '
+                                                    'Microsoft.OSTCExtensions.DummyExtension-1.2.3.4')
+
+        self.assertEqual(1, mock_warn.call_count)
+        self.assertEqual(0, mock_error.call_count)
+
+    @patch("azurelinuxagent.common.logger.StdoutAppender.write")
+    @patch("azurelinuxagent.common.logger.ConsoleAppender.write")
+    @patch("azurelinuxagent.common.event.send_logs_to_telemetry", return_value=True)
+    def test_telemetry_logger_verify_maximum_recursion_depths_doesnt_happen(self, *_):
+        logger.add_logger_appender(logger.AppenderType.FILE, logger.LogLevel.INFO, path="/dev/null")
+        logger.add_logger_appender(logger.AppenderType.TELEMETRY, logger.LogLevel.WARNING, path=add_log_event)
+
+        for i in range(MAX_NUMBER_OF_EVENTS):
+            logger.warn('Test Log - {0} - 1 - Warning'.format(i))
+
+        exception_caught = False
+
+        # #1035 was caused due to too many files being written in an error condition. Adding even one more here broke
+        # the camels back earlier. This should be resolved now.
```

Comment:
can we clarify in the comment that this would go into an infinite recursion that would raise a RuntimeError?